x-pack/plugins/observability_solution/slo/public/embeddable/slo/error_budget/error_budget_react_embeddable_factory.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+import React, { useEffect } from 'react';
+import { Router } from '@kbn/shared-ux-router';
+import { createBrowserHistory } from 'history';
+import { ReactEmbeddableFactory } from '@kbn/embeddable-plugin/public';
+import {
+  initializeTitles,
+  useStateFromPublishingSubject,
+  fetch$,
+} from '@kbn/presentation-publishing';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { SLO_ERROR_BUDGET_ID } from './constants';
+import { SloErrorBudgetEmbeddableState, SloEmbeddableDeps, ErrorBudgetApi } from './types';
+import { SloErrorBudget } from './error_budget_burn_down';
+
+export const getErrorBudgetPanelTitle = () =>
+  i18n.translate('xpack.slo.errorBudgetEmbeddable.title', {
+    defaultMessage: 'SLO Error Budget burn down',
+  });
+
+export const getErrorBudgetEmbeddableFactory = (deps: SloEmbeddableDeps) => {
+  const factory: ReactEmbeddableFactory<SloErrorBudgetEmbeddableState, ErrorBudgetApi> = {
+    type: SLO_ERROR_BUDGET_ID,
+    deserializeState: (state) => {
+      return state.rawState as SloErrorBudgetEmbeddableState;
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+      const defaultTitle = getErrorBudgetPanelTitle();
+      titlesApi.setPanelTitle(state.title ?? defaultTitle);
+      const sloId$ = new BehaviorSubject(state.sloId);
+      const sloInstanceId$ = new BehaviorSubject(state.sloInstanceId);
+      const reload$ = new Subject<boolean>();
+
+      const api = buildApi(
+        {
+          ...titlesApi,
+          serializeState: () => {
+            return {
+              rawState: {
+                ...serializeTitles(),
+                sloId: sloId$.getValue(),
+                sloInstanceId: sloInstanceId$.getValue(),
+              },
+            };
+          },
+        },
+        {
+          sloId: [sloId$, (value) => sloId$.next(value)],
+          sloInstanceId: [sloInstanceId$, (value) => sloInstanceId$.next(value)],
+          ...titleComparators,
+        }
+      );
+
+      const fetchSubscription = fetch$(api)
+        .pipe()
+        .subscribe((next) => {
+          reload$.next(next.isReload);
+        });
+      return {
+        api,
+        Component: () => {
+          const sloId = useStateFromPublishingSubject(sloId$);
+          const sloInstanceId = useStateFromPublishingSubject(sloInstanceId$);
+          const I18nContext = deps.i18n.Context;
+          const queryClient = new QueryClient();
```

Comment:
This `QueryClient` class will be created on every render of this component. Is that intended? If not, you could move this outside of the Component, then it will only be created when the embeddable is built.

Reply:
We need `QueryClient` cause we use a hook to get the error budget history data that uses `useQuery`. If I remove it, whole Kibana breaks. I moved it out of the Component as suggested.